app/controllers/base_channel.js

```diff
@@ -16,14 +16,20 @@ export default class BaseChannelController extends Controller {
 
   @alias('application.showChannelMenu') showChannelMenu;
 
-  createMessage (message, type) {
-    return new Message({
+  createMessage (content, type) {
+    const message = new Message({
       type: type,
       date: new Date(),
       // TODO  nickname per channel
       nickname: this.model.account.nickname,
-      content: message
+      content: content
     });
+
+    if (this.model.protocol === 'XMPP') {
```

Comment:
Why only for XMPP? Wouldn't it be the same process for IRC?

Reply:
Nope. You don't receive your own messages back from a channel in IRC.